MeshLib/MeshEditing/FlipElements.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "FlipElements.h"
+
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Line.h"
+#include "MeshLib/Elements/Tri.h"
+#include "MeshLib/Elements/Quad.h"
+#include "DuplicateMeshComponents.h"
+
+namespace MeshLib
+{
+
+MeshLib::Element* flipElement(MeshLib::Element const& elem, std::vector<MeshLib::Node*> const& nodes)
+{
+	if (elem.getDimension()>2)
+		return nullptr;
+
+	std::size_t const n_nodes (elem.getNNodes());
+	MeshLib::Node** elem_nodes = new MeshLib::Node*[n_nodes];
+	for (unsigned i=0; i<n_nodes; ++i)
+		elem_nodes[i] = nodes[elem.getNode(i)->getID()];
+	MeshLib::Node* tmp (elem_nodes[0]);
+	elem_nodes[0] = elem_nodes[1];
+	elem_nodes[1] = tmp;
+
+	if (elem.getGeomType() == MeshElemType::LINE)
```

Comment:
If you used switch, there would be warnings generated if at some point in time somebody extends the enum.